contrib/babelfishpg_tds/src/backend/tds/tdsutils.c

```diff
@@ -707,9 +706,6 @@ tdsutils_ProcessUtility(PlannedStmt *pstmt,
 		case T_DropdbStmt:
 			handle_result = handle_dropdb((DropdbStmt *) parsetree);
 			break;
-		case T_AlterRoleStmt:
-			handle_result = handle_alter_role((AlterRoleStmt*)parsetree);
```

Comment:
Even in the above scenario, the unprivileged TSQL user will not have sysadmin membership and eventually they won't have CREATEROLE privilege to ALTER any Babelfish role from PG port.

Reply:
Is that a valid scenario that user can drop babelfishpg_tsql but keep babelfishpg_tds?
Tried this: [babelfishpg_tds is dependent on babelfishpg_tsql, so user cannot drop babelfishpg_tsql only]
```
jdbc_testdb=# drop extension babelfishpg_tsql;
ERROR:  cannot drop extension babelfishpg_tsql because other objects depend on it
DETAIL:  view master_dbo.sysdatabases depends on view sys.sysdatabases
view tempdb_dbo.sysdatabases depends on view sys.sysdatabases
view msdb_dbo.sysdatabases depends on view sys.sysdatabases
extension babelfishpg_tds depends on extension babelfishpg_tsql
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
jdbc_testdb=# drop extension babelfishpg_tsql cascade;
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to view master_dbo.sysdatabases
drop cascades to view tempdb_dbo.sysdatabases
drop cascades to view msdb_dbo.sysdatabases
drop cascades to extension babelfishpg_tds
ERROR:  could not open relation with OID 18589
```
If this is the case, handling PG dialect for sysadmin role here makes more sense to me. Otherwise we would need to define the repeated helper function in TDS as well.

Reply:
Another case was that if user is connected to a different PG database and tries to alter the babelfish role from there, in that case since there isn't any babelfish extension installed it will allow altering babelfish role.